src/kafka/kafka_clients.clj

```diff
@@ -0,0 +1,99 @@
+(ns kafka.kafka-clients
+  (:import (java.net InetAddress)
+           (java.util Properties)
+           (org.apache.kafka.clients.producer KafkaProducer ProducerRecord)
+           (org.apache.kafka.common.serialization StringSerializer StringDeserializer)
+           (org.apache.kafka.clients.consumer KafkaConsumer)
+           (java.time Duration)))
+
+(defn build-producer-config
+  []
+  (let [producer-config (new Properties)]
+    (.putAll producer-config {"value.serializer"  StringSerializer
+                              "key.serializer"    StringSerializer
+                              "client.id"         (.getHostName (InetAddress/getLocalHost))
+                              "bootstrap.servers" "localhost:9092"
+                              "acks"              "all"})
+    producer-config))
```

Comment:
For clarity, in your `let` forms only the final return value should be outside of the binding blocks `[]`